models/comments.models.js

```diff
@@ -1,12 +1,37 @@
 const db = require("../db/connection");
-const { checkArticleExists } = require("../utils/check-exists");
+const { checkExists } = require("../utils/check-exists");
 
 exports.selectCommentsByArticle = async (article_id) => {
   try {
-    await checkArticleExists(article_id)
-    const { rows } = await db.query("SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at ASC", [article_id]);
+    await checkExists("articles", "article_id", article_id, "Article");
+    const { rows } = await db.query(
+      "SELECT * FROM comments WHERE article_id = $1 ORDER BY created_at ASC",
+      [article_id]
+    );
     return rows;
   } catch (err) {
     return Promise.reject(err);
   }
 };
+
+exports.insertCommentByArticle = async (article_id, newComment) => {
+  const { username, body } = newComment;
+  if (!body) return Promise.reject({ status: 400, msg: "Empty Comment" });
+  try {
+    await checkExists("articles", "article_id", article_id, "Article");
+    await checkExists("users", "username", username, "User");
```

Comment:
As the author column on the comment is a foreign key, psql will throw an error if we attempt to post a user not found on the user table. As a general rule, always let psql throw an error if it's going to and a custom error if it doesn't. Try processing this error through the query and see what happens 👍